File: app/src/main/java/com/github/ymegane/android/taptaptap/domain/usecase/SoundEffectUseCaseImpl.java
package com.github.ymegane.android.taptaptap.domain.usecase;

import android.content.Context;

import com.github.ymegane.android.taptaptap.data.repository.SoundEffectRepositoryImpl;
import com.github.ymegane.android.taptaptap.domain.repository.SoundEffectRepository;

public class SoundEffectUseCaseImpl implements SoundEffectUseCase {

    private final SoundEffectRepository repository;

    public SoundEffectUseCaseImpl(SoundEffectRepository repository) {
        this.repository = repository;
        this.repository.load();
    }

    public SoundEffectUseCaseImpl(Context context) {
        this.repository = new SoundEffectRepositoryImpl(context);
        this.repository.load();
    }


    @Override
    public void playRandom() {
        repository.play();
    }

    @Override
    public void stop() {
       repository.release();
    }
}

File: app/src/main/java/com/github/ymegane/android/taptaptap/domain/usecase/SoundEffectUseCaseImpl.kt
package com.github.ymegane.android.taptaptap.domain.usecase

import android.content.Context

import com.github.ymegane.android.taptaptap.data.repository.SoundEffectRepositoryImpl
import com.github.ymegane.android.taptaptap.domain.repository.SoundEffectRepository

class SoundEffectUseCaseImpl : SoundEffectUseCase {

    private val repository: SoundEffectRepository

    @Suppress("unused")
    constructor(repository: SoundEffectRepository) {
        this.repository = repository
        this.repository.load()
    }

    constructor(context: Context) {
        this.repository = SoundEffectRepositoryImpl(context)
        this.repository.load()
    }

    override fun playRandom() = repository.play()

    override fun stop() = repository.release()
}
